llama_index/vector_stores/postgres.py

```diff
@@ -362,13 +362,20 @@ def _build_sparse_query(
         limit: int,
         metadata_filters: Optional[MetadataFilters] = None,
     ) -> Any:
-        from sqlalchemy import select
+        from sqlalchemy import select, type_coerce
         from sqlalchemy.sql import func, text
+        from sqlalchemy.types import UserDefinedType
+
+        class REGCONFIG(UserDefinedType):
+            def get_col_spec(self, **kw: Any) -> str:
+                return "regconfig"
```

Comment:
plainto_tsquery expects its first argument to be of type regconfig, but sqlalchemy has to add casts with asyncpg, so the query that gets sent is actually SELECT plainto_tsvector($1::varchar, $2::varchar), which was causing errors due to the wrong type.

They fixed this bug in 2.0 https://github.com/sqlalchemy/sqlalchemy/issues/8977 also adding REGCONFIG type, but since we wanted to support 1.4, we had to add that type manually and then cast the first argument of plaintto_tsvector. This was suggested here: https://github.com/sqlalchemy/sqlalchemy/issues/6833#issuecomment-892844560